src/Spice86.Core/Emulator/Memory/Indexer/SegmentedAddressTupleIndexer.cs

```diff
@@ -0,0 +1,25 @@
+namespace Spice86.Core.Emulator.Memory.Indexer;
+
+/// <summary>
+/// Retrieves Segment / Offset pairs stored in Memory.
+///
+/// Offset is the first value and Segment the second.
+/// This layout is common for various instructions / interrupt table / ...
+/// Does not instantiates an object, returns a tuple.
+/// </summary>
+public class SegmentedAddressTupleIndexer : Indexer<(ushort Segment, ushort Offset)> {
```

Comment:
Maybe rename this SegmentAddressValueIndexer, and the SegmentAddressTuple property to SegmentAddressValue.

This reflects the fact there tuples are value types, therefore they usually live on the satck. Which means a different lifetime from  SegmentAddres objects (but we have the same equality comparison behavior).

In the .NET BCL, there is the same kind of separation (Task vs ValueTask, Tuple vs ValueTuple, ...)

Plus if the values inside the tuple are not meant to be modified, this could be a ``` record struct ``` instead.

Reply:
Done for the renaming, for record struct it could be nice indeed in order not to rely on param orders. But that would be another PR :)